URDF model generation with all the distributed inertial sensors

nunoguedelha opened the pull request:
This is the final part of changes for supporting the automatic generation of URDF-like and Gazebo-like sensor elements in the URDF models, as a complement of the PRs https://github.com/robotology/simmechanics-to-urdf/pull/22 and https://github.com/robotology/gazebo-yarp-plugins/pull/304.

In the YAML configuration files, the section under the tag `sensors` has been populated with the link sensors parameters described [here](https://github.com/robotology/simmechanics-to-urdf#sensors-parameters). For each sensor added in that section, the parameters `exportFrameInURDF` and `updateRate` have the following default values:

- `exportFrameInURDF`: "No", actually, there is no more need to export the frames, since the sensor frames are automatically added as additional frames when creating a `iDynTree::Model` from an URDF file (refer to this [discussion](https://github.com/robotology-playground/icub-model-generator/issues/20#issuecomment-239698424))
- `updateRate`: "100", already previously used
- `sensorBlobs`: Gazebo plugin selection for handling the sensor

So for each specific sensor, we just define the parameters `frameName`, `linkName`, `sensorName`, `sensorType` ([detailed description here](https://github.com/robotology/simmechanics-to-urdf#sensors-parameters)).

nunoguedelha commented:
@traversaro , I had added the `.ini` files under the folder `../conf/MTB/` but later realised that all the config `ini` files will probably be defined only in the `icub-models` repo. I guess we will later cleanup all the config files in a row once a clear choice has been made regarding the configuration for the Gazebo simulation.

nunoguedelha commented:
@traversaro , I realised that the tool `gz` generating the **sdf** file converts the prefix "package://" into "model://" for the meshes but not for the *gazebo yarp plugins* (probably because these are used exclusively by Gazebo). And as Gazebo requires the prefix "model://" for finding the `ini` files, the model loading fails.
I've corrected the issue, and will push it after a few more tests.

traversaro commented on simmechanics/data/icub2_5/conf/MTB/gazebo_icub_torso_mtb.ini:
```diff
@@ -0,0 +1,21 @@
+enabledSensors (iCub::chest::chest_mtb_acc_9b7
+				iCub::chest::chest_mtb_acc_9b8
+				iCub::chest::chest_mtb_acc_9b9
+				iCub::chest::chest_mtb_acc_9b10)
```
Not blocking, but for the future: always avoid tabs, because you never know with which width they are rendered (for example, I guess in your editor they are visualized with four spaces, while on GitHub they are visualized as 8 spaces.

traversaro commented:
That's great. 

Let me know if you want to generate and commit the models yourself in `icub-models` or not.

nunoguedelha commented:
@traversaro , just : 
- added the ini files
- fixed the Gazebo plugins path prefix (`package` -> `model`)
- fixed torso MTB inertial sensors IDs
- definition of the `imu_frame` as we discussed last time (IMU no more defined in the sensor list)

traversaro commented:
Ok, if the PR is ready to be merged please remove the `[WIP]` prefix from the title, thanks!

nunoguedelha commented:
Sorry, I forgot about that :)

nunoguedelha commented:
I have already a branch [feature/updateModels](https://github.com/nunoguedelha/icub-models/tree/feature/updateModels) with the `iCubGenova04` model updated in my fork of the repo. I just need to add the `model.config` for the other models, do a few checks and pull request it. Will do it tonight after the summer school course.
We can discuss later if you want, in case you have some other requirement in mind.

traversaro commented:
I think `icub-models` should just contain file that are automatically copied/generated by the target `copy-models-to-icub-models` of `icub-model-generator`. If that part is not complete (missing `model.config`, etc, etc) no problem we can always add them later.
